package/src/activation.ts

```diff
@@ -0,0 +1,462 @@
+        pos: sourcegraph.Position
+    ): AsyncGenerator<
+        sourcegraph.Badged<sourcegraph.Hover> | null | undefined,
+        void,
+        undefined
+    > {
+        const lsifResult = await lsifProviders.hover(doc, pos)
+        if (lsifResult) {
+            yield lsifResult.value
+            return
+        }
+
+        if (lspProviders) {
+            yield* lspProviders.hover(doc, pos)
+            return
+        }
+
+        const searchResult = await searchProviders.hover(doc, pos)
+        if (!searchResult) {
+            yield undefined
+            return
+        }
+
+        yield { ...searchResult, badge: impreciseBadge }
+    }
+
+    return {
+        provideHover: memoizePrevious(
+            areProviderParamsEqual,
+            (textDocument, position) =>
+                observableFromAsyncGenerator(
+                    abortPrevious(() => provideHover(textDocument, position))
+                ).pipe(shareReplay(1))
+        ),
+    }
+}
+
+function registerExternalReferencesProvider<S extends { [key: string]: any }>(
+    ctx: sourcegraph.ExtensionContext,
+    selector: sourcegraph.DocumentSelector,
+    externalReferencesProvider: ExternalReferenceProvider
+) {
+    const settings: BehaviorSubject<Partial<S>> = new BehaviorSubject<
+        Partial<S>
+    >({})
+    ctx.subscriptions.add(
+        sourcegraph.configuration.subscribe(() => {
+            settings.next(sourcegraph.configuration.get<Partial<S>>().value)
+        })
+    )
+
+    let registration: sourcegraph.Unsubscribable | undefined
+
+    const register = () => {
+        registration = sourcegraph.languages.registerReferenceProvider(
+            selector,
+            createExternalReferencesProvider(externalReferencesProvider)
+        )
+    }
+
+    const deregister = () => {
+        if (registration) {
+            registration.unsubscribe()
+            registration = undefined
+        }
+    }
+
+    ctx.subscriptions.add(
+        from(settings)
+            .pipe(
+                map(
+                    settings =>
+                        !!settings[externalReferencesProvider.settingName]
+                ),
+                distinctUntilChanged(),
+                map(enabled => (enabled ? register : deregister)()),
+                finalize(() => deregister())
+            )
+            .subscribe()
+    )
+}
+
+function createExternalReferencesProvider(
+    externalReferencesProvider: ExternalReferenceProvider
+): sourcegraph.ReferenceProvider {
+    return {
+        provideReferences: (doc, pos, ctx) =>
+            // TODO - add memoizePrevious, abortPrevious
+            observableFromAsyncGenerator(() =>
+                externalReferencesProvider.references(doc, pos, ctx)
+            ),
+    }
+}
+
+function registerImplementationsProvider(
+    ctx: sourcegraph.ExtensionContext,
+    selector: sourcegraph.DocumentSelector,
+    implementationsProvider: ImplementationsProvider
+) {
+    ctx.subscriptions.add(
```

Comment:
Logic for this method was taken from the go and typescript extensions, which differed only on the impl id and panel titles.